### ISSUE TITLE
allow indexing into components

### ISSUE THREAD

**Issue by mathisonian** [idyll-lang/idyll#30]
As discussed on gitter, from @rreusser:

It would be nice to be able to index into components, to allow for sharing scope for related components.

For example:
```
[Slideshow.Container]
  [Slideshow.Slide]x[/Slideshow.Slide]
  [Slideshow.Slide]x[/Slideshow.Slide]
  [Slideshow.Slide]x[/Slideshow.Slide]
[/Slideshow.Container]
```

In this case, the `Slideshow.Container` component should resolve to `require('slideshow').Container`

**Comment by rreusser** [comment]
👍 Seems like it would make it so much easier to create reusable pieces that work *together*.

**Comment by mathisonian** [comment]
Closed by #45
